### PR TITLE
EN-46931: Fix window function check

### PR DESCRIPTION
**Pull request by chitang**
Aggregate functions can have window info

### CODE REVIEW

File: soql-analyzer/src/main/scala/com/socrata/soql/typechecker/Typechecker.scala [hunk 1]
```diff
@@ -161,6 +158,19 @@ class Typechecker[Type](typeInfo: TypeInfo[Type], functionInfo: FunctionInfo[Typ
     }
   }
 
+  private def checkWindowFunction(fc: FunctionCall, candidateFn: MonomorphicFunction[_]): Unit = {
```
Comment (saraid) [inline soql-analyzer/src/main/scala/com/socrata/soql/typechecker/Typechecker.scala:161]: Is this always going to be a `MonomorphicFunction`? Why not `Function` here?

Comment (chitang) [reply soql-analyzer/src/main/scala/com/socrata/soql/typechecker/Typechecker.scala:161]: We are hitch-hiking on the resolved funtions which are always Monomorphic.  We don't need another way to figure out the Functions(s).

Comment (saraid) [reply soql-analyzer/src/main/scala/com/socrata/soql/typechecker/Typechecker.scala:161]: Good enough for me.

**Review by saraid — APPROVED** [review]
Seems to cover the relevant cases.

**Review by saraid — APPROVED** [review]
Yeah, that's eleganter.

**Pull request merged**